### PR TITLE
806 added second hook

### PR DESCRIPTION
**Pull request by kevinsallee**
* the first default hook just returns the first file it finds
* the second default hook inherits from the first one, but returns a published file only if it matches one of the viewer extensions, in order. If not, it raises a TankError

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def _launch_viewer(self, path):
         if system.startswith("linux"):
             cmd = '%s "%s" &' % (app_path, path)
         elif system == "darwin":
-            cmd = 'open -n "%s" --args "%s"' % (app_path, path)
+            cmd = 'open -a "%s" "%s"' % (app_path, path)
```
Comment (gplsteph) [inline app.py]: Why did you change this? The `-n` option forces a new app to be used, without it a running app (e.g. Maya) will be used to load the file.

File: app.py [hunk 2]
```diff
@@ -35,9 +35,10 @@ def context_change_allowed(self):
     def init_app(self):
         deny_permissions = self.get_setting("deny_permissions")
         deny_platforms = self.get_setting("deny_platforms")
+        title = self.get_setting("display_name")
```
Comment (gplsteph) [inline app.py]: Default value?

File: hooks/shotgun_get_published_file_for_viewer.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+If none are found, raise a TankError.
+"""
+
+import sgtk
+from sgtk import TankError
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class GetPublishedFileForViewer(HookBaseClass):
```
Comment (gplsteph) [inline hooks/shotgun_get_published_file_for_viewer.py:26]: Should the class have a different name?

**Review by gplsteph — CHANGES_REQUESTED** [review]
Not approving because of the `open -a` changes and the fact that there is no default value for the display name.
I'm a bit confused as well: it seems the default logic is back to what it used to be, which might be perfectly fine, but I just wanted to double check it was intended and that if the default implementation should prefer viewer extensions if available, like you had it previously?

File: app.py (hunk 1 above)
Comment (kevinsallee) [reply app.py]: open -n Photoshop --args `blah` seems to work for PSDs only. Any other document type, it just opens photoshop but nothing else.

File: hooks/shotgun_get_published_file_for_viewer.py (hunk 3 above)
Comment (kevinsallee) [reply hooks/shotgun_get_published_file_for_viewer.py:26]: Like what?

File: app.py (hunk 2 above)
Comment (kevinsallee) [reply app.py]: what do you mean?

File: hooks/shotgun_get_published_file_for_viewer.py (hunk 3 above)
Comment (gplsteph) [reply hooks/shotgun_get_published_file_for_viewer.py:26]: Not sure, may be the base class should be called `GetPublishedFile`? Not sure, and not a biggy, it just feel strange to have the same class names for both.

File: app.py (hunk 2 above)
Comment (gplsteph) [reply app.py]: `title = self.get_setting("display_name", "Open in Associated Application")`? Is the display_name setting mandatory?

File: hooks/shotgun_get_published_file_for_viewer.py (hunk 3 above)
Comment (kevinsallee) [reply hooks/shotgun_get_published_file_for_viewer.py:26]: one is called `GetPublishedFile` and the other `GetPublishedFileForViewer`. I don't get it.

Comment (gplsteph) [reply hooks/shotgun_get_published_file_for_viewer.py:26]: Fine if it is the case, but this is what I'm seeing
Screen Shot 2019-09-11 at 16 25 52

File: app.py (hunk 1 above)
Comment (kevinsallee) [reply app.py]: what should I do? I read the documentation for `open`, tried a bunch of things, only `open -a` seems to work

**Comment by gplsteph** [comment]
Quite sure this will have unforeseen effects and will never be accepted on SG master branch. Sorry I don’t have a practical solution

Stéphane Deverly

Le 11 sept. 2019 à 21:02, Kevin Sallee <notifications@github.com<mailto:notifications@github.com>> a écrit :


@kevinsallee commented on this pull request.

________________________________

In app.py<https://github.com/GPLgithub/tk-shotgun-launchpublish/pull/2#discussion_r323409237>:

> @@ -90,7 +90,7 @@ def _launch_viewer(self, path):
         if system.startswith("linux"):
             cmd = '%s "%s" &' % (app_path, path)
         elif system == "darwin":
-            cmd = 'open -n "%s" --args "%s"' % (app_path, path)
+            cmd = 'open -a "%s" "%s"' % (app_path, path)


what should I do? I read the documentation for open, tried a bunch of things, only open -a seems to work

File: app.py (hunk 1 above)
Comment (gplsteph) [reply app.py]: what about 'open -n -a "%s" "%s"'?

**Review by gplsteph — CHANGES_REQUESTED** [review]
Not approving just for the missing '-n' param. It seems '-n -a XXXX file' works.
